src/S3Bucket/BucketDisplay.vue

```diff
@@ -0,0 +1,182 @@
+        <CreateModal :api="api" :resource="resource" :state="state" @update:refreshResource="refreshResource"/>
+        <VBtn v-if="isFlat" icon="mdi-folder-eye" title="Toggle Folder View" size="x-large" @click="fetchFlattenedView"/>
+        <VBtn v-else icon="mdi-view-headline"  title="Toggle Flat List View" size="x-large" @click="fetchFlattenedView" />
+      </VBtnGroup>
+      <VTooltip location="top" text="Toggle Version Mode" >
+        <template #activator="{ props: activatorProps }">
+          <VSwitch v-if="hasVersionMode" v-model="isVersionMode" v-bind="activatorProps" inset color="primary" label="Version" density="compact" hide-details class="flex-grow-0 mr-2"/>
+        </template>
+      </VTooltip>
+    </div>
+      <VAlert v-if="dropError" closable type="warning" icon="mdi-alert-circle" text="Cannot upload Folders. Drop files only." />
+      <VCard ref="dropZoneRef" flat>
+        <NestedTable 
+          :api="api"
+          :location="location"
+          :resource="resource"
+          :is-loading="isLoading"
+          :state="state"
+          :is-version-mode="isVersionMode"
+          :data-table-items="dataTableItems"
+          :selected-items="selectedItems"
+          :fetch-selection="fetchSelection"
+          :refresh-resource="refreshResource"
+          @update:items="(val) => selectedItems = val"
+          />
+        <VOverlay v-model="isOverDropZone" contained class="align-center justify-center text-h6 text-white">Drop Files to Upload</VOverlay>
+    </VCard>
+  </div>
+</template>
+
+<script setup>
+import { useDropZone } from '@vueuse/core';
+import { computed, onUnmounted, ref } from "vue";
+import { convertObjectToFormData } from '../helpers/axiosHelper';
+import BucketBreadcrumbs from "./Components/BucketBreadcrumbs.vue";
+import DownloadButton from "./Components/DownloadButton.vue";
+import NestedTable from "./Components/NestedTable.vue";
+import CreateModal from "./Modals/CreateModal.vue";
+import DeleteModal from "./Modals/DeleteModal.vue";
+import MultiFileModal from './Modals/MultiFileModal.vue';
+import UploadModal from "./Modals/UploadModal.vue";
+
+/**
+ * @typedef {Object} Props
+ * @property {ReturnType<import("@cloudbolt/js-sdk").createApi>} Props.api - The authenticated API instance
+ * @property {String} Props.location - The selected S3 Bucket location
+ * @property {Object} Props.resource - The S3 Bucket resource
+ * @property {Object} Props.state - The selected S3 Bucket state
+ * @property {Function} Props.updateResourceSelection - Function to replace The S3 Bucket resource
+ */
+/** @type {Props} */
+const props = defineProps({
+  api: {
+    type: Object,
+    required: true,
+  },
+  location: {
+    type: String,
+    default: '',
+  },
+  resource: {
+    type: Object,
+    default: () => {},
+  },
+  state: {
+    type: Object,
+    default: () => {},
+  },
+  updateResourceSelection: {
+    type: Function,
+    default: () => {},
+  },
```

Comment:
You generally don't want to pass functions in as props - that's what Events are for! How could you refactor this (and all the other places where you pass in functions) to use events instead? https://vuejs.org/guide/essentials/event-handling.html

Reply:
In this particular case, this function is passing data back down to the parent. If that's the case, I might suggest using vue's two-way data with v-model. https://vuejs.org/guide/components/v-model.html

Reply:
👍

Reply:
👍

Reply:
FWIW, we pivoted again in a zoom meeting and are using a composable instead.